pydantic/fields.py

```diff
@@ -630,7 +630,7 @@ def _type_analysis(self) -> None:  # noqa: C901 (ignore complexity)
             self.key_field = self._create_sub_type(get_args(self.type_)[0], 'key_' + self.name, for_keys=True)
             self.type_ = get_args(self.type_)[1]
             self.shape = SHAPE_DEFAULTDICT
-        elif issubclass(origin, Dict):
+        elif origin is dict or origin is Dict:
```

Comment:
Maybe: 
```suggestion
        elif issubclass(origin, (dict, Dict)):
```
?

Reply:
One additional piece of info: it looks like `type(Base.__fields__["b"].outer_type_)` is `<class 'types.GenericAlias'>`, but `type(Sub.__fields__["b"].outer_type_)` is `<class 'dict'>`. 🤔

It looks like maybe `.outer_type_` is getting replaced with some other dictionary *instance* somehow in subclass creation?
```py
(Pdb) m.__fields__["b"].outer_type_.keys()
dict_keys([4336358208, 4340970304, 4341193536, 4341045632, 4341212736, 4340612832, 4341193344, 4341212480, 4341239232, 4341200384, 4341200448, 4340970688, 4341193856])
(Pdb) m.__fields__["b"].outer_type_.values()
*** AttributeError: name
```

Reply:
Hi @JacobHayes 
Thanks for the review
So yes what didn't work before is 
```py
m = Base(a=frozendict(a=1), b=frozendict(b='1'))
assert isinstance(m.a, frozendict)
assert isinstance(m.b, frozendict)
assert m.b['b'] == 1  # coerced
```
This is now resolved because the `frozendict` is not coerced in `dict`.
The 'name' issue is something else indeed. I'll fix this as well

Reply:
In case you're interested, this escalated to a cPython bug with `types.GenericAlias`: https://bugs.python.org/issue45167